python/tchannel/event.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import
+from enum import IntEnum
+import logging
+from .exceptions import InternalEventException
+
+log = logging.getLogger('tchannel')
+
+
+class EventType(IntEnum):
+    send_request = 0x00,
+    send_response = 0x01,
+    receive_request = 0x02,
+    receive_response = 0x03
+
+
+class EventHook(object):
+    """provide all event hook interfaces
+
+    Customized Hook should should inherit from EventHook class and implement
+    the events' hooks that it wants to listen.
+
+    Example::
+        TraceHook(EventHook):
+            def send_request(self, context):
+                ....
+
+    """
+
+    def __init__(self, priority=0):
+        self.priority = priority
+
+    def __cmp__(self, other):
+        return cmp(self.priority, other.priority)
```

Comment:
You don't need cmp on something that's supposed to represent a computation of some kind.